test/XmlHttpRequestConnection.tests.js

```diff
@@ -145,6 +145,36 @@ define(function (require) {
 
         });
 
+        describe("is correctly using XmlHttpRequest and runs attached callbacks while performing:", function (){
+            beforeEach(function (){
+                mockXMLHttpRequest.prototype.send = function (body) {
+                    this.readyState = 4;
+                    this.status = 200;
+                    this.onloadstart();
+                    this.onreadystatechange();
+                };
+                spyOn(mockXMLHttpRequest.prototype, 'send').andCallThrough();
+
+                window.XMLHttpRequest = (function () { return mockXMLHttpRequest; }());
+
+                connection = new XmlHttpRequestConnection();
+            });
+
+            it("execute call and invoke onloadstart callback", function () {
```

Comment:
since you insist in https://github.com/ezsystems/ez-js-rest-client/pull/83#discussion_r106599156 on keeping a very strict list of events, you need to test each (and also the test for `upload` event handlers are missing)